compiler-plugin-tests/src/test/java/io/ballerina/stdlib/graphql/compiler/CompilerPluginTest.java

```diff
@@ -144,12 +144,19 @@ public void testGraphQLListTypeInputParametersInResources() {
     }
 
     @Test
-    public void testGraphQLSingleFileUpload() {
+    public void testGraphQLFileUpload() {
         String packagePath = "valid_service_15";
         DiagnosticResult diagnosticResult = getDiagnosticResult(packagePath);
         Assert.assertEquals(diagnosticResult.errorCount(), 0);
     }
 
+    @Test
+    public void testIntersectionTypes() {
+        String packagePath = "valid_service_16";
+        DiagnosticResult diagnosticResult = getDiagnosticResult(packagePath);
+        Assert.assertEquals(diagnosticResult.errorCount(), 0);
+    }
+
```

Comment:
Shall we add a test for a service using the same intersection type as input and return type?

Reply:
I thought the use of input as an output (and vice-versa) is handled by the specific test cases. Do we need separate tests for that in this scenario?

Reply:
In those test cases, we didn't use intersection types as I remember.

Reply:
Yes, but the point is, input-output type checking is done separately, without considering whether it is an intersection type or not. So, if the same type is used as an input and as an output, it should throw an error regardless of it being an intersection type, right?